variants/github_actions_ci/workflows/ci.yml

```diff
@@ -0,0 +1,132 @@
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: postgres
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+      # Uncomment these if you need them for your test suite
+      # redis:
+      #   image: redis:latest
+      #   ports:
+      #     - 6379:6379
+      # elasticsearch:
+      #   image: elasticsearch:latest
+      #   env:
+      #     cluster.name: whats-on-test-cluster
+      #     discovery.type: single-node
+      #     ES_JAVA_OPTS: "-Xms1g -Xmx1g"
+      #   ports:
+      #     - 9200:9200
+      #     - 9300:9300
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+      - name: Install Required OS Packages
+        run: sudo apt-get -yqq install libpq-dev
+      - name: Set up Ruby (version given by .ruby-version)
+        uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+      - run: bundle exec rubocop
+      - run: bundle exec brakeman --run-all-checks --exit-on-warn --format plain .
+      - run: bundle exec bundle audit --update
+      - run: bundle exec rails db:setup
+      - run: bundle exec rspec spec --format progress
+
+  # ACTION REQUIRED:
+  #   Uncomment the deployment block which suits your application. You should
+  #   remove all code (in this file and other yaml files in this dir) which
+  #   relate to other kinds of deployment.
+  #
+  # ######################################################################### #
+  # Deploy to EC2 with Capistrano
+  # ######################################################################### #
+  # deploy_to_ec2_staging:
+  #   if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+  #   needs:
+  #     - ruby_based_checks
+  #     - js_based_checks
+  #   uses: ./.github/workflows/deploy_to_ec2.yml
+  #   with:
+  #     environment: staging
+  #     assume_role_arn: TODO # e.g. "arn:aws:iam::<ACCOUNT_ID>:role/<CLIENT_NAME>StagingGHARole"
+  #     aws_region: ap-southeast-2 # Sydney
+  #   secrets:
+  #     slack_webhook: ${{ secrets.SLACK_WEBHOOK }}
+  #     ssh_private_key: ${{ secrets.STAGING_SSH_PRIVATE_KEY }}
+  # deploy_to_ec2_production:
+  #   if: github.event_name == 'push' && github.ref == 'refs/heads/production'
+  #   needs:
+  #     - ruby_based_checks
+  #     - js_based_checks
+  #   uses: ./.github/workflows/deploy_to_ec2.yml
+  #   with:
+  #     environment: production
+  #     assume_role_arn: TODO # e.g. "arn:aws:iam::<ACCOUNT_ID>:role/<CLIENT_NAME>ProductionGHARole"
+  #     aws_region: ap-southeast-2 # Sydney
+  #   secrets:
+  #     slack_webhook: ${{ secrets.SLACK_WEBHOOK }}
+  #     ssh_private_key: ${{ secrets.PRODUCTION_SSH_PRIVATE_KEY }}
+
+  # ######################################################################### #
+  # Deploy to Heroku
+  # ######################################################################### #
```

Comment:
I'm tempted to have a `deploy_to_heroku` variant now just so we can test for it and uncomment this if it is enabled. It could also add a `.herokuignore` file I guess. Anyway that's a bit out of scope for this PR so I'll leave it alone here.

Reply:
We could also have it create the `Procfile`, and maybe delete `staging.rb` configs since Heroku doesn't like those?

I think now that we've got this file having a heroku variant definitely makes sense (but agree lets do it in a follow up PR)